cylc/flow/scripts/cylc_scan.py

```diff
@@ -59,267 +158,356 @@ def get_option_parser():
+                name: flow[key]
+                for name, key in (
+                    ('version', 'cylcVersion'),
+                    ('host', Cont.HOST),
+                    ('port', Cont.PORT)
+                )
+            }
+        }
+        maxlen = max(len(key) for key in display)
+        for key, value in display.items():
+            # format multiline strings by whitespace padding the lines
+            value = ('\n' + (' ' * (maxlen + 7))).join(value.splitlines())
+            # write out the key: value pairs
+            ret.append(f'    {key: <{maxlen}}   {value}')
+    return '\n'.join(ret)
+
+
+def sort_function(flow):
+    if flow.get('status') == 'running':
+        state = 0
+    elif flow.get('status') == 'held':
+        state = 0
+    elif flow.get('contact'):
+        state = 2
+    else:
+        state = 3
+    return (state, flow['name'])
+
+
+async def _sorted(pipe, formatter, opts, write):
+    """List all flows, sort, then print them individually."""
+    ret = []
+    async for item in pipe:
+        ret.append(item)
+    for flow in sorted(ret, key=sort_function):
+        write(formatter(flow, opts))
+
+
+async def _serial(pipe, formatter, opts, write):
+    """List all flows, then print them as one."""
+    ret = []
+    async for item in pipe:
+        ret.append(item)
+    write(formatter(ret, opts))
+
+
+async def _async(pipe, formatter, opts, write):
+    """List and print flows individually."""
+    async for flow in pipe:
+        write(formatter(flow, opts))
+
+
+async def _tree(pipe, formatter, opts, write):
+    """List all flows, sort, then print them as a tree."""
+    # get list of flows
+    ret = []
+    async for flow in pipe:
+        ret.append(flow)
+
+    # construct tree
+    tree = {}
+    for flow in sorted(ret, key=lambda f: f['name']):
+        parts = Path(flow['name']).parts
+        pointer = tree
+        for part in parts[:-1]:
+            if part not in pointer:
+                pointer[part] = {}
+            pointer = pointer[part]
+        flow['name'] = parts[-1]
+        item = formatter(flow, opts)
+        if len(parts) > 1:
+            item = f' {item}'
+        pointer[item] = ''
+
+    # print tree
+    ret = get_tree(tree, '', sort=False, use_unicode=True)
+    write('\n'.join(ret))
+
+
+def get_pipe(opts, formatter, scan_dir=None):
+    """Construct a pipe for listing flows."""
+    if scan_dir:
+        pipe = scan(scan_dir=scan_dir)
+    else:
+        pipe = scan
+
+    show_running = 'running' in opts.states
+    show_held = 'held' in opts.states
+    show_active = show_running or show_held or 'stopping' in opts.states
+    # show_active = bool({'running', 'held'} & opts.states)
+    show_inactive = bool({'stopped'} & opts.states)
+
+    # filter by flow name
+    if opts.name:
+        pipe |= filter_name(*opts.name)
+
+    # filter by flow state  -  TODO: reduce this
+    if show_active and show_inactive:
+        pipe |= is_active(True, filter_stop=False)
```

Comment:
It's a combo of `show_active` and `show_inactive` and if `not show_active and not show_inactive` the `is_active` step is skipped entirely.

Reply:
```
    if show_active and show_inactive:
        pipe |= is_active(True, filter_stop=False)
    elif show_active:
        pipe |= is_active(True)
    elif show_inactive:
        pipe |= is_active(False)
```
Reduce to:
```
    if show_active:
        pipe |= is_active(True, filter_stop=(not show_inactive))
    elif show_inactive:
        pipe |= is_active(False)
```
Since the default for `filter_stop` must be `True`